Instant Search: Replace color literals at build time

sgomes opened the pull request:
Currently, the entire `@automattic/color-studio` color configuration is being bundled into the Instant Search main bundle, even though only two values are used out of that set.

This PR instead replaces the color values being used with literals at build-time. This maintains all the benefits of relying on the external package for configuration and having the entire set at one's disposal, while bringing the Instant Search main bundle size down by about 4KB (1.5KB gzipped). All it takes is a little build-time magic, with `webpack`'s built-in `DefinePlugin`.

#### Changes proposed in this Pull Request:
* Replace color literals at build time in Instant Search

#### Jetpack product discussion
None.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Ensuring that the build completes successfully should be enough. Some smoke-testing that the Jetpack colophon is still rendered using the correct colors is probably a good idea as well.

#### Proposed changelog entry for your changes:
No need for a changelog entry; this should be a seamless under-the-hood change. That said, this and other ongoing performance improvements to Instant Search could probably be grouped into a single entry along the lines of: 
* Instant Search: improved performance by reducing JavaScript file size.

sgomes committed:
Bundle colors at build time in Instant Search

jsnmoon reviewed (dismissed):
👍 Works as described! Colophon renders as expected and size reduction is significant (46.05 kB gzipped -> 44.61 gzip kB gzipped).

sgomes committed:
Add changelog entry

sgomes commented:
Even though this PR preceded the changelogger requirement, it was still being blocked by the changelogger merge check.

I rebased and added a changelogger entry, but unfortunately this reset the review flag, which means I still can't merge. Would you mind re-approving, @jsnmoon? Thank you!

sgomes commented:
Thank you for the review, @jsnmoon! 👍

sgomes commented:
Changeset ID 222830.
